jerry-core/jerryscript.h

```diff
@@ -344,6 +344,13 @@ size_t jerry_parse_and_save_literals (const jerry_char_t *source_p, size_t sourc
                                       uint32_t *buffer_p, size_t buffer_size, bool is_c_format);
 
 /**
+ * Module management
+ */
+int jerry_register_module(void);
+void *jerry_get_module_data(int index);
+void jerry_set_module_data(int index, void *data, void (*deleter)(void *));
```

Comment:
global static, assigned only once when the constructor gets called (implemented ideally via `__attribute__((constructor))` but an exported function that is called from `main()` will work too).